.github/workflows/build-ui-and-server.yml

```diff
@@ -155,9 +158,43 @@ jobs:
           directory: ./provider-ui/coverage/lcov-report
           flags: integrationtests
           fail_ci_if_error: true
+
+  run-lint:
+    name: Eslint Check
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v3
+
+    - uses: actions/setup-node@v3.0.0
+      with:
+        node-version: ${{ secrets.NODE_VERSION }}
+
+    - name: Get npm cache directory
+      id: npm-cache-dir
+      run: |
+        cd ui
+        echo "::set-output name=dir::$(npm config get cache)"
+
+    - uses: actions/cache@v3
+      id: npm-cache
+      with:
+        path: ${{ steps.npm-cache-dir.outputs.dir }}
+        key: ${{ runner.os }}-node-${{ hashFiles('ui/package-lock.json') }}
+        restore-keys: |
+          ${{ runner.os }}-node-
+
+    - name: Install modules
+      if: steps.npm-cache.outputs.cache-hit != 'true'
+      run: make setup-ui-libs
+
+    - name: Run ESLint
+      run: |
+        make lint-ui
+        make lint-provider-ui
+
```

Comment:
This eslint should be a seperate github action, it shouldn't be build into this. This is already very complicated since it has integration and e2e tests already.